### PR TITLE
Feature/fga/rx flow migration

### LINKED ISSUE #4219: Finish migration from RxJava to Flow
Finish migration from RxJava to Flow in the application.
Will allow to remove the -rx artefact for Mavericks.

### PR DESCRIPTION
**Pull request by ganfra**
Closes #4219 
Please test it carefully.
I made some profiling to track stuff being done on main thread, looking good.

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 1]
```diff
@@ -864,11 +865,13 @@ class RoomDetailViewModel @AssistedInject constructor(
     private fun observeEventDisplayedActions() {
         // We are buffering scroll events for one second
         // and keep the most recent one to set the read receipt on.
-        visibleEventsObservable
-                .buffer(1, TimeUnit.SECONDS)
+
+        visibleEventsSource
+                .stream()
+                .chunk(1000)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:871]: Yes

File: dependencies.gradle [hunk 2]
```diff
@@ -115,13 +115,13 @@ ext.libs = [
                 'bigImageViewer'         : "com.github.piasy:BigImageViewer:$bigImageViewer",
                 'glideImageLoader'       : "com.github.piasy:GlideImageLoader:$bigImageViewer",
                 'progressPieIndicator'   : "com.github.piasy:ProgressPieIndicator:$bigImageViewer",
-                'glideImageViewFactory'  : "com.github.piasy:GlideImageViewFactory:$bigImageViewer"
+                'glideImageViewFactory'  : "com.github.piasy:GlideImageViewFactory:$bigImageViewer",
+                'flowBinding'            : "io.github.reactivecircus.flowbinding:flowbinding-android:$flowBinding",
+                'flowBindingAppcompat'   : "io.github.reactivecircus.flowbinding:flowbinding-appcompat:$flowBinding",
+                'flowBindingMaterial'    : "io.github.reactivecircus.flowbinding:flowbinding-material:$flowBinding"
```
[the remark this answers is not in the document]
Comment (ganfra) [reply dependencies.gradle:121]: Yep I will

File: vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt [hunk 3]
```diff
@@ -166,12 +165,12 @@ class DevicesViewModel @AssistedInject constructor(
 //                    )
 //                }
 
-        refreshPublisher.throttleFirst(4_000, TimeUnit.MILLISECONDS)
-                .subscribe {
+        refreshSource.stream().sample(4_000)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt:168]: Not sure this is really important but changed

File: vector/src/main/assets/open_source_licenses.html [hunk 4]
```diff
@@ -279,25 +279,9 @@ <h3>
...
-        <b>rxrelay</b>
-        <br/>
-        Copyright 2014 Netflix, Inc.
-        Copyright 2015 Jake Wharton
-    </li>
-    <li>
-        <b>rxbinding</b>
-        <br/>
-        Copyright (C) 2015 Jake Wharton
+        Copyright 2019 Yang Chen
```
Comment (bmarty) [inline vector/src/main/assets/open_source_licenses.html:284]: 👍

**Review by bmarty — APPROVED** [review]
Thanks for the update
For Element 1.3.8

**Pull request merged**